lib/if.c

```diff
@@ -219,6 +219,18 @@ struct interface *if_lookup_by_index(ifindex_t ifindex, vrf_id_t vrf_id)
 	struct vrf *vrf;
 	struct interface if_tmp;
 
+	if (vrf_id == VRF_UNKNOWN) {
+		struct interface *ifp;
+
+		RB_FOREACH(vrf, vrf_id_head, &vrfs_by_id) {
```

Comment:
Observation -- this will find the first instance of a nexthop across all the vrf's -- if there are multiple instances (say 10.1.1.1 exists in two differen vrfs) it will only find the first. I think this is okay, and I don't know what else you would do here, just wanted to bring it up to make certain this case is considered.

Reply:
This is a lookup by ifindex not an ip address, and ifindex's are unique in the system so I think we are good

Reply:
Ah, yes -- this makes sense. This is good then.

Reply:
ifindexes won't be unique when we add support for using netns's as VRFs :/

Reply:
that sucks :(